plugin/init.lua

```diff
@@ -60,35 +63,82 @@ local function get_file_path(file_name, type, opt_name)
 	return string.format("%s%s" .. separator .. "%s.json", pub.save_state_dir, type, file_name:gsub(separator, "+"))
 end
 
+---@alias encryption_opts {enable: boolean, private_key: string | nil, public_key: string | nil, encrypt: fun(file_path: string, lines: string[]), decrypt: fun(file_path: string): string | nil}
+pub.encryption = {
+	enable = false,
+	private_key = nil,
+	public_key = nil,
+	encrypt = function(file_path, lines)
+		local success, stdout, stderr = wezterm.run_child_process({
```

Comment:
I am not too keen on the arguments to this child process, but I have not managed to find a better way to do this. Please let me know if you know an alternative way of doing this @joncrangle

Reply:
This implementation has the benefit of keeping data encrypted at rest and only encrypts/decrypts when absolutely needed. Other options I can think of would be less secure because the state would remain decrypted for longer periods of time.

I just submitted a PR to this branch with updates I needed to make to get it working in my config + some doc content.

Reply:
We should stick to that implementation then